File: Array/026/026.py
class Solution(object):

    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        用一个列表将需要删除的值存起来，然后遍历一次进行删除
        """
        nList = []
        if len(nums) > 1:
            for i in range(len(nums) - 1):
                if nums[i] == nums[i + 1]:
                    nList.append(nums[i])
            for x in nList:
                nums.remove(x)
        return len(nums)

File: Array/026/026.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function removeDuplicates(nums) {
  let len = 1; // 最少有一个元素不重复
  for (let i=1; i<nums.length; i++) { // 从第2个元素开始跟前面一个比较
    if (nums[i] !== nums[i-1]) {
      nums[len++] = nums[i]; // 不等于前一个元素的话长度加1，并将新元素加到前len个
    }
  }
  return len;
}
